tests/Api/Parser/JsonParserTest.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+use Aws\Api\Parser\Exception\ParserException;
+use Aws\Api\Service;
+use Aws\AwsClient;
+use GuzzleHttp\Promise\FulfilledPromise;
+use GuzzleHttp\Psr7\Response;
+use PHPUnit\Framework\TestCase;
+
+/*
+ * This class tests the PHP specific functionality of the JsonParser;
+ * cross-SDK testing is done in ComplianceTest.php
+ */
+
+/**
+ * @covers \Aws\Api\Parser\JsonRpcParser
+ * @covers \Aws\Api\Parser\JsonParser
+ */
+class JsonParserTest extends TestCase
+{
+    public function testUnixTimestampReturnedFromService()
+    {
+        $service = $this->generateTestService();
+        $client = $this->generateTestClient($service, ['timestamp' => 932169600]);
+        $command = $client->getCommand('ParseJson');
+        $list = $client->getHandlerList();
+        $handler = $list->resolve();
+        $result = $handler($command)->wait()['timestamp']->__toString();
+
+        self::assertEquals("1999-07-17T00:00:00+00:00", $result);
+    }
+    public function testTextDateReturnedFromService()
```

Comment:
Perhaps `testTextDateWhenModeledAsIso8601`? And so on, for other tests. Although, if you end up using the data provider pattern, they'd end up under an umbrella test anyway.